livedata/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -97,14 +97,15 @@ class ChannelControllerImpl(
     val unfilteredMessages: LiveData<List<Message>> = Transformations.map(_messages) { it.values.toList() }
 
     /** a list of messages sorted by message.createdAt */
-    override val messages: LiveData<List<Message>> = Transformations.map(_messages) {
+    override val messages: LiveData<List<Message>> = Transformations.map(_messages) { messageMap ->
         // TODO: consider removing this check
-        it.values
+        val messages = messageMap.values
             .asSequence()
             .filter { it.parentId == null || it.showInChannel }
             .filter { hideMessagesBefore == null || it.createdAt!! > hideMessagesBefore }
             .sortedBy { it.createdAt }
             .toList()
+        messages.map { it.copy(cid = cid) }
```

Comment:
```suggestion
        messageMap.values
            .asSequence()
            .filter { it.parentId == null || it.showInChannel }
            .filter { hideMessagesBefore == null || it.createdAt!! > hideMessagesBefore }
            .sortedBy { it.createdAt }
            .toList()
            .map { it.copy(cid = cid) }
```
We can join both lines

Reply:
👍